From 2b60b1bb918ae1579949e71839141f9eb09b170d Mon Sep 17 00:00:00 2001
From: Robert Sowula <robert@sowula.at>
Date: Wed, 1 Nov 2023 21:06:54 +0100
Subject: [PATCH] Migrate deprecated imp package to importlib

---
 setup.py | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 60a2a2a..1f5948d 100644
--- a/setup.py
+++ b/setup.py
@@ -1,8 +1,18 @@
 from setuptools import setup, find_packages

-import imp
+import importlib.util
+import importlib.machinery

-version = imp.load_source('jams.version', 'jams/version.py')
+
+def load_source(modname, filename):
+    loader = importlib.machinery.SourceFileLoader(modname, filename)
+    spec = importlib.util.spec_from_file_location(modname, filename, loader=loader)
+    module = importlib.util.module_from_spec(spec)
+    loader.exec_module(module)
+    return module
+
+
+version = load_source('jams.version', 'jams/version.py')

 setup(
     name='jams',
